### ISSUE TITLE
Support new GAV of plexus-build-api

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#944]
See https://github.com/codehaus-plexus/plexus-build-api/commit/9c2e01325564b5154bea04b2c913ec016ef71db4

https://github.com/codehaus-plexus/plexus-build-api/issues/21

We could simply have both APIs supported.

**Comment by HannesWell** [comment]
> See [codehaus-plexus/plexus-build-api@9c2e013](https://github.com/codehaus-plexus/plexus-build-api/commit/9c2e01325564b5154bea04b2c913ec016ef71db4)
> 
> [codehaus-plexus/plexus-build-api#21](https://github.com/codehaus-plexus/plexus-build-api/issues/21)
> 
> We could simply have both APIs supported.

The dependency can be added/changed here (just in case you didn't already know :) ):
https://github.com/eclipse-m2e/m2e-core/blob/4df09e69ba221bf7db7dceeac9e5ed666dfb5606/m2e-maven-runtime/org.eclipse.m2e.maven.runtime/pom.xml#L74

If the fully-qualified names did not change we have to or can use that as drop in replacement.

**Comment by kwin** [comment]
@laeubi Is something still missing in https://github.com/eclipse-m2e/m2e-core/compare/master...laeubi:m2e-core:issue_944_new_plexus_api or can we include this in the upcoming release?

**Comment by laeubi** [comment]
It is at least missing that I need to finish/finalize that :-)

**Comment by laeubi** [comment]
@gnodet @kwin if you like to speed this up, the most critical part (for me) currently is that we do not have good test-coverage but high impact if we break things.
So if we can have some test in https://github.com/eclipse-m2e/m2e-core/tree/master/org.eclipse.m2e.core.tests that cover the functionality this would help a lot, I already tried before getting an integration-test with a dummy-maven plugin but can't get it reliable working that a plugin compiled in the current reactor is used in a test.

**Comment by gnodet** [comment]
@laeubi the branch does not build anymore from the command line.
I end up with 
```
[ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.0-SNAPSHOT:compare-version-with-baselines (check-no-version-regression) on project org.eclipse.m2e.model.edit: Version have moved backwards for (org.eclipse.m2e.model.edit/2.0.200.20230130-0743). Baseline has 2.0.300.20230320-2130) with delta: 0.0.-100 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.0-SNAPSHOT:compare-version-with-baselines (check-no-version-regression) on project org.eclipse.m2e.model.edit: Version have moved backwards for (org.eclipse.m2e.model.edit/2.0.200.20230130-0743). Baseline has 2.0.300.20230320-2130) with delta: 0.0.-100
```
but there's no references to either version in the source tree.

**Comment by gnodet** [comment]
Also and fwiw, the project does not load correctly in IntelliJ (all the Tyco projects are not recognised) nor Eclipse Java (class path does not seem to be initialised correctly).

I've done quite a lof to OSGi development in the past, but not using Tycho, Eclipse or M2e, so I'm not familiar with the whole setup.

**Comment by laeubi** [comment]
M2E is an eclipse project so Eclipse IDE is somehow a requirement (even though it *might* work to some degree with IntelliJ).

You can setup a Dev-IDE here:
https://github.com/eclipse-m2e/m2e-core/blob/master/CONTRIBUTING.md#%EF%B8%8F-setting-up-the-development-environment-automatically-using-the-eclipse-installer-oomph

Regarding the branch, the best would be to just start from the master and as a first step add a test-case before try to apply any of the changes this can then be merged already as a prerequisite and will show that the dev-workflow works already.

If we have this I can rebase my branch ontop of this to see if it already breaks something, and add support for the new GAV.

**Comment by laeubi** [comment]
It should actually already work as there is a redirect from one API to the other.
@mickaelistria du you face any specific problem here?

**Comment by mickaelistria** [comment]
> du you face any specific problem here?

Not really, I just stumbled upon some related discussions that are still open as Maven moves some tracking to GitHub issues ( https://github.com/apache/maven-assembly-plugin/issues/1150 ). Note that the "how to make Maven plugin m2e-friendly" doc still mentions the sonatype variant of BuildContext. Should it be updated to use the plexus one?

**Comment by laeubi** [comment]
> Should it be updated to use the plexus one?

Sounds reasonable.

**Comment by wilx** [comment]
What is the status of the compatibility for the Plexus API classes vs the old Sonatype API classes in the m2e? Can this be closed because https://github.com/eclipse-sisu/sisu-project/pull/215 was merged?

**Comment by laeubi** [comment]
I'm not sure what sisu has to do with it but the new plexus-api does contain a compatibility layer and therfore m2e can use the old+new API
